File: ts-playwright/src/pages/CartPage.ts
import { Locator, Page } from '@playwright/test';

export class CartPage {
  readonly page: Page;
  readonly items: Locator;
  readonly checkoutButton: Locator;

  constructor(page: Page) {
    this.page = page;
    this.items = page.locator('.cart_item');
    this.checkoutButton = page.locator('#checkout');
  }

  async itemNames(): Promise<string[]> {
    return this.page.locator('.cart_item .inventory_item_name').allInnerTexts();
  }

  async checkout(): Promise<void> {
    await this.checkoutButton.click();
  }
}

File: python-playwright/framework/pages/cart_page.py
"""SauceDemo cart page."""

from __future__ import annotations

from playwright.sync_api import Page

from .base_page import BasePage


class CartPage(BasePage):
    def __init__(self, page: Page) -> None:
        super().__init__(page)
        self.items = page.locator(".cart_item")
        self.checkout_button = page.locator("#checkout")

    def item_names(self) -> list[str]:
        return self.page.locator(".cart_item .inventory_item_name").all_inner_texts()

    def checkout(self) -> None:
        self.checkout_button.click()
